### PR TITLE
API Integration

### LINKED ISSUE #37: [FR] Connect API
# Connect API

### Description of the feature
* Install Axios to make RESTful (for now) requests to the API

### Purpose of the feature
* FInally implement the backend along with front

### Tasks to complete
- [ ] Install axios
- [ ] Turn all template data into requests
- [ ] Test

### PR DESCRIPTION
**Pull request by cadrianxyz**
# Description

API endpoints integration with axios. Work in progress - validation not done. All endpoints not tested yet.
> Netlify not building due to console.errors

Fixes #37 

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. If design change, please ensure design change/addition has been approved by a UX Lead or 

- [x] Unit-Testing (not jest)

# Checklist:

- [x] Code follows the project's style guidelines (need lint merges from master)
- [x] I have performed a self-review of the Code being pushed
- [ ] Code has been reviewed by at least one other developer
- [x] Code has appropriate commenting, particularly in hard-to-understand areas
- [x] Corresponding changes to the documentation have been made
- [x] The changes generate no new warnings

### CODE REVIEW

**Comment by cadrianxyz** [comment]
Good to go except for event initiation page - logic is kinda weird so its kinda hard. will definitely need more time.

IF this PR is merged, will need to start using the api along with this to actually 'see' events on the map and load them in `eventDetails`

File: src/views/EventPage.vue [hunk 1]
```diff
@@ -86,22 +87,31 @@ export default {
...
+      snipped: true,
+
+      backgroundImg: null,
+      numPhotos: 0,
+      imageList: [],
+      tags: [],
+      eventName: '',
+      venueName: '',
+      startDate: '',
+      description: '',
```
Comment (kraytaps) [inline src/views/EventPage.vue:101]: i think we should just make all these event details one whole object

Comment (cadrianxyz) [inline src/views/EventPage.vue:101]: to sort of wrap it this way?
```suggestion
      event: {
          backgroundImg: null,
          numPhotos: 0,
          imageList: [],
          tags: [],
          eventName: '',
          venueName: '',
          startDate: '',
          description: '',
      }
```

**Comment by cadrianxyz** [comment]
@kraytaps PR is ready for review. Recommend merging [event-data branch](https://github.com/onlinkers/event-discovery-api/pull/23) first so that when setting up mongo, you can initialize using template data first. Otherwise, an empty mongodb database will not return any events

**Review by kraytaps — APPROVED** [review]

**Pull request merged**